Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Migration/Tools/MigrationWindow.cs

```diff
@@ -0,0 +1,244 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.UI;
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+using Object = UnityEngine.Object;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Utilities
+{
+    /// <summary>
+    /// This is an Utility Window for the MigrationTool. 
+    /// </summary>
+    public class MigrationWindow : EditorWindow
+    {
+        private enum ToolbarOption
+        {
+            GameObjects = 0,
+            Scenes = 1,
+            Project = 2
+        };
+
+        private static readonly string[] toolbarTitles =
+        {
+            "Game Objects",
+            "Scenes",
+            "Full Project"
+        };
+
+        private ToolbarOption selectedToolbar = ToolbarOption.GameObjects;
+        private Vector2 scrollPosition = Vector2.zero;
+
+        private const string MigrationWindowURL = "https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Tools/MigrationWindow.html";
+        private const string WindowTitle = "Migration Window";
+        private const string WindowDescription = "This tool allows the migration of obsolete components into up-to-date versions.";
+
+        private int selectedMigrationHandlerIndex;
+        private string[] migrationHandlerTypeNames;
+        private bool isMigrationEnabled;
+        private Type selectedMigrationHandlerType;
+
+        private readonly MigrationTool migrationTool = new MigrationTool();
+
+        [MenuItem("Mixed Reality Toolkit/Utilities/Migration Window", false, 4)]
+        private static void ShowWindow()
+        {
+            var window = GetWindow<MigrationWindow>(typeof(SceneView));
+            window.titleContent = new GUIContent(WindowTitle, EditorGUIUtility.IconContent("d_TimelineEditModeRippleOFF").image);
+            window.minSize = new Vector2(400.0f, 300.0f);
+            window.Show();
+        }
+
+        private void OnEnable()
+        {
+            isMigrationEnabled = false;
+            migrationTool.ClearMigrationList();
+
+            // Adds empty as first option for MigrationHandler selector 
+            var migrationTypeNamesList = new List<string> { "" };
+            migrationTypeNamesList.AddRange(migrationTool.MigrationHandlerTypes
+                                  .Select(x => x.FullName)
+                                  .ToList());
+            migrationHandlerTypeNames = migrationTypeNamesList.ToArray();
```

Comment:
Seems unnecessary?

```
migrationHandlerTypeNames = migrationTool.MigrationHandlerTypes
                                  .Select(x => x.FullName)
                                  .ToArray();
```

Reply:
MigrationTypeNames needs an extra item. The first item should be empty so we can have an empty option on the popup selector. Suggestions?

Reply:
hmmm, I see. nvm